### PR TITLE
machines: Fix some issues in nic edit dialog

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by KKoukiou** [comment]
> Looks reasonable, thank you!

[root@localhost ~]# brctl addbr br1 && virsh attach-interface --current subVmTest1 bridge br1
-bash: brctl: command not found


on rhel images.. Putting needswork

File: test/verify/check-machines-dbus [hunk 1]
```diff
@@ -292,6 +295,60 @@ class TestMachinesDBus(machineslib.TestMachines):
...
+
+        # Save the network settings
+        b.click("#vm-subVmTest1-network-1-edit-dialog-save")
+        b.wait_not_present("#vm-subVmTest1-network-1-edit-dialog-modal-window")
+
+        b.wait_in_text("#vm-subVmTest1-network-1-type", "network")
+        b.wait_in_text("#vm-subVmTest1-network-1-source", "test_network")
+
+        # bridge-utils are not available on RHEL test images
+        if m.image not in ["rhel-8-0-distropkg", "rhel-8-1"]:
```
Comment (martinpitt) [inline test/verify/check-machines-dbus]: This needs "rhel-8-0" as well then. But please file a PR to rebuild the images with bridge-utils.

Comment (KKoukiou) [reply test/verify/check-machines-dbus]: Sure, I am on it.

File: test/verify/check-machines-dbus [hunk 2]
```diff
@@ -292,6 +295,60 @@ class TestMachinesDBus(machineslib.TestMachines):
...
+            m.execute("virsh net-destroy test_network && virsh net-destroy default && virsh net-undefine test_network && virsh net-undefine default")
+
+            # Remove the network interface
+            m.execute("virsh detach-interface --persistent --type network --domain subVmTest1")
+
+            # Start the VM in order that the UI picks up the new interface, since we don't get events of shut off domains
+            m.execute("virsh start subVmTest1")
+
+            # Create a second bridge to LAN NIC
+            m.execute("brctl addbr br1 && virsh attach-interface --current subVmTest1 bridge br1")
```
Comment (martinpitt) [inline test/verify/check-machines-dbus]: As we've seen in #11840, let's not use the obsolete brctl here, but the `bridge` command.

Comment (KKoukiou) [reply test/verify/check-machines-dbus]: Sure, I was not aware that it's not used any-more.
Replaced the command in the tests. Sorry for the confusion.

**Comment by martinpitt** [comment]
[This testNetworkSettings](https://209.132.184.41:8493/logs/pull-11833-20190516-113737-0e3c1a6b-verify-debian-testing/log.html#197) failure might be relevant? [Looks like](https://209.132.184.41:8493/logs/pull-11833-20190516-113737-0e3c1a6b-verify-debian-testing/TestMachinesDBus-testNetworkSettings-debian-testing-127.0.0.2-2301-FAIL.png) the "default" nework is still active here. Marking needswork for now.

**Comment by KKoukiou** [comment]
I see https://209.132.184.41:8493/logs/pull-11833-20190516-113737-0e3c1a6b-verify-debian-testing/log.html#197 failure, needs work again :(

**Comment by KKoukiou** [comment]
> [This testNetworkSettings](https://209.132.184.41:8493/logs/pull-11833-20190516-113737-0e3c1a6b-verify-debian-testing/log.html#197) failure might be relevant? [Looks like](https://209.132.184.41:8493/logs/pull-11833-20190516-113737-0e3c1a6b-verify-debian-testing/TestMachinesDBus-testNetworkSettings-debian-testing-127.0.0.2-2301-FAIL.png) the "default" nework is still active here. Marking needswork for now.

Looks like a race since I didn't wait somewhere for the networks to actually get removed from the app state.
Added a one liner condition for that, hopefully it will solve the problem.

**Pull request merged**